### PR TITLE
Also check part scores for corruptions

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #15977

Since this may increase both the length and the amount of the error messages, I've additionally made the error dialog resizable when "Show details" is clicked, so that the user always has a way to see the full message.

https://user-images.githubusercontent.com/48658420/213683303-9f7ad8fa-201d-4039-b19f-750128167403.mov

### CODE REVIEW

**Comment by bkunda** [comment]
@cbjeukendrup I might suggest we use a horizontal scroll bar (on hover) instead of making the dialog resizable. We should already have a vertical scroll bar where the number of error messages exceeds the available height of the frame, and I think that offering a resizable error dialog is perhaps less discoverable, especially since we don't (as far as I'm aware) use that option for any other dialog of this nature in the app.

**Comment by cbjeukendrup** [comment]
@bkunda I added a horizontal scrollbar. Unless you disagree, I would still prefer to make the dialog resizable in addition, because especially for scores with many corruptions, it is quite uncomfortable if you need to scroll through all those dozens of lines in that tiny amount of space.

**Comment by bkunda** [comment]
> @bkunda I added a horizontal scrollbar. Unless you disagree, I would still prefer to make the dialog resizable in addition, because especially for scores with many corruptions, it is quite uncomfortable if you need to scroll through all those dozens of lines in that tiny amount of space.

I don't have a massive problem with the resizing option, although I'm not sure it will be necessary; as far as I'm aware, for cases where there exist many corruptions, the primary application of this "show more" section is to allow the user to copy the error messages and paste them elsewhere, so that the errors can then be manually fixed; the user can't address these errors with the dialog open, so being able to expand the dialog to see more of the errors has limited practicality (Although one can of course screenshot the expanded dialog).

So it's fine to have it resizable; I just thought I'd add this point as I'm not sure it's really that necessary (but perhaps I'm also overlooking something here).

**Comment by Eism** [comment]
I also don't see the point in resizing the dialog.
@cbjeukendrup  let's remove the possibility of resizing since the UX designer also does not see the point in this + there is no such possibility in standard Qt dialogs with details section.

**Comment by cbjeukendrup** [comment]
Okay, I've removed it...

Makes the "details" view quite annoying to interact with though, because scrolling through so many rows in such a small area is just annoying. But yeah, most of the time it will be useless anyway because currently you'd need to copy & paste it somewhere else anyway if you want to see it while working on the score...

**Review by Eism — APPROVED** [review]

File: src/engraving/libmscore/check.cpp [hunk 1]
```diff
@@ -106,34 +106,65 @@ void Score::checkScore()
...
-                for (size_t v = 0; v < VOICES; ++v) {
-                    EngravingItem* element = s->element(static_cast<int>(staffIdx) * VOICES + static_cast<int>(v));
+                for (voice_idx_t v = 0; v < VOICES; ++v) {
+                    EngravingItem* element = s->element(staffIdx * VOICES + v);
                     if (!element) {
                         continue;
                     }
 
                     ChordRest* cr = toChordRest(element);
-                    if (cr == 0) {
```
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/engraving/libmscore/check.cpp:134]: We already check `if (!element)` above. And `toChordRest` is not like `dynamic_cast`.

**Comment by bkunda** [comment]
🟢 Re-tested in macOS. FIXED.

**Pull request merged**